osu.Game/Screens/SelectV2/ICarouselPanel.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+
+namespace osu.Game.Screens.SelectV2
+{
+    /// <summary>
+    /// An interface to be attached to any <see cref="Drawable"/>s which are used for display inside a <see cref="Carousel{T}"/>.
+    /// </summary>
+    public interface ICarouselPanel
+    {
+        /// <summary>
+        /// The Y position which should be used for displaying this item within the carousel.
+        /// </summary>
+        double YPosition => Item!.CarouselYPosition;
+
+        /// <summary>
+        /// The carousel item this drawable is representing. This is managed by <see cref="Carousel{T}"/> and should not be set manually.
+        /// </summary>
+        CarouselItem? Item { get; set; }
```

Comment:
This setter being public-public looks rather bad, but there don't seem to be any elegant ways of hiding it (i.e. ones that don't involve casting), so maybe something I'd look away from...

Reply:
I tried multiple alternatives and couldn't find a nicer way without making things overly complicated.